specification/sbi_atee.adoc

```diff
@@ -101,35 +101,75 @@ This operation requires the host to convert a sufficient number of non-confident
+     * the documentation of `sbi_tee_host_run_tvm_vcpu()` for more details.
+     */
+    uint64_t csrs[1024];
+};
+
+struct tsm_shmem_scratch {
+    /* General purpose registers for a TVM guest.
+     *
+     * The TSM will always read or write the minimum number of registers in this set to complete
+     * the requested action. To avoid leaking information from the TVM, the TSM must follow the
+     * given rules.
+     *
+     * The TSM will write to these registers upon return from `sbi_tee_host_run_tvm_vcpu()` when:
+     *  - The vCPU takes a store guest page fault in an emulated MMIO region.
+     *  - The vCPU makes an ECALL that is to be forwarded to the host.
+     *
+     * The TSM will read from these registers when:
+     *  - The vCPU takes a load guest page fault in an emulated MMIO region.
+     */
+    uint64_t guest_gprs[32];
+    uint64_t _reserved[224];
+};
+-------------------
 
 ===== VCPU creation
 The host must register CPUs/harts with the TSM before they can be used for TVM execution by calling
-`sbi_tee_host_create_tvm_vcpu()`. The host must also configure the non-confidential shared memory
-that's set-up by the host while creating the VCPU. The shared memory is used both the host and the
-TSM for when processing TVM exits from `sbi_tee_run_vcpu()`.
+`sbi_tee_host_create_tvm_vcpu()`. The NACL shared memory interface is used between the host and the
+TSM for processing TVM exits from `sbi_tee_host_run_tvm_vcpu()`.
 
 ===== TVM execution
 Following the assignment of memory and VCPU resources, the host can transition the guest into a `TVM_RUNNABLE`
 state by calling `sbi_tee_host_finalize_tvm()`. Note that some TEE calls are no longer permissible after this
 transition.
 
-The host can use the aforementioned shared-memory to set up TVM execution parameters like the
-entrypoint (`ENTRY_PC`) / boot argument (`ENTRY_ARG`), then `sbi_tee_host_finalize_tvm()`, followed by
-sbi_tee_host_run_tvm_vcpu()` to begin execution. TVM execution continues until there an event like an
-interrupt, or fault that cannot be serviced by the TSM. Some interrupts and exceptions are resumable,
-and the host can determine reason specific reason by examining the `scause` field in the `tvm_vcpu_supervisor_csrs`
-previously setup by the call to `sbi_tee_host_create_tvm_vcpu()`. The host can then examine the shared-memory region
-if needed to determine further course of action. This may involve servicing exits caused by TVM-ECALLs that require
-host action(like adding of MMIO and shared-memory regions), TVM page-faults, virtual instructions, etc.
+The host can then call sbi_tee_host_run_tvm_vcpu()` to begin execution. TVM execution continues until there is 
+an event like an interrupt, or fault that cannot be serviced by the TSM. Some interrupts and exceptions are resumable,
+and the host can determine reason specific reason by examining the `scause` CSR in the NACL shared memory. The host
```

Comment:
The host reads scause/stval directly where reads htinst and htval from the NACL shared memory.